tests/screenshots/build.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+use std::io::Write;
+use std::path::Path;
+
+/// Returns a list of all the `.slint` files in the `tests/cases` subfolders.
+pub fn collect_test_cases() -> std::io::Result<Vec<test_driver_lib::TestCase>> {
+    let mut results = vec![];
+
+    let case_root_dir: std::path::PathBuf = [env!("CARGO_MANIFEST_DIR"), "cases"].iter().collect();
+
+    println!("cargo:rerun-if-env-changed=SLINT_TEST_FILTER");
+    let filter = std::env::var("SLINT_TEST_FILTER").ok();
+
+    for entry in walkdir::WalkDir::new(case_root_dir.clone()).follow_links(true) {
+        let entry = entry?;
+        let absolute_path = entry.into_path();
+        if absolute_path.is_dir() {
+            println!("cargo:rerun-if-changed={}", absolute_path.display());
+            continue;
+        }
+        let relative_path =
+            std::path::PathBuf::from(absolute_path.strip_prefix(&case_root_dir).unwrap());
+        if let Some(filter) = &filter {
+            if !relative_path.to_str().unwrap().contains(filter) {
+                continue;
+            }
+        }
+        if let Some(ext) = absolute_path.extension() {
+            if ext == "60" || ext == "slint" {
+                results.push(test_driver_lib::TestCase { absolute_path, relative_path });
+            }
+        }
+    }
+    Ok(results)
+}
+
+fn main() -> std::io::Result<()> {
+    let mut generated_file = std::fs::File::create(
+        Path::new(&std::env::var_os("OUT_DIR").unwrap()).join("generated.rs"),
+    )?;
+
+    let references_root_dir: std::path::PathBuf =
+        [env!("CARGO_MANIFEST_DIR"), "references"].iter().collect();
+
+    for (i, testcase) in collect_test_cases()?.into_iter().enumerate() {
+        let case_root_dir: std::path::PathBuf =
+            [env!("CARGO_MANIFEST_DIR"), "cases"].iter().collect();
+
+        let reference_path = testcase.absolute_path.strip_prefix(case_root_dir).unwrap();
+        let mut reference_path = references_root_dir
+            .join(reference_path)
+            .with_extension("png")
+            .into_os_string()
+            .into_string()
+            .unwrap();
+
+        reference_path = format!("\"{}\"", reference_path);
+
+        println!("cargo:rerun-if-changed={}", testcase.absolute_path.display());
+        let mut module_name = testcase.identifier();
+        if module_name.starts_with(|c: char| !c.is_ascii_alphabetic()) {
+            module_name.insert(0, '_');
+        }
+        writeln!(generated_file, "#[path=\"{0}.rs\"] mod r#{0};", module_name)?;
+        let source = std::fs::read_to_string(&testcase.absolute_path)?;
+
+        let mut output = std::fs::File::create(
+            Path::new(&std::env::var_os("OUT_DIR").unwrap()).join(format!("{}.rs", module_name)),
+        )?;
+
+        if !generate_macro(&source, &mut output, testcase)? {
```

Comment:
You should not use the macro test because it doesn't use the image embedding.

Reply:
Then I need to call the compiler right?

Reply:
right, there was a `generate_source` function that did that.